pkg/clusteragent/admission/mutate/autoinstrumentation/config_test.go

```diff
@@ -41,8 +41,28 @@ func TestNewInstrumentationConfig(t *testing.T) {
 		{
 			name:       "config with extra fields errors",
 			configPath: "testdata/extra_fields.yaml",
-			shouldErr:  true,
-			expected:   nil,
+			shouldErr:  false,
+			expected: &InstrumentationConfig{
```

Comment:
Really appreciate this change @keisku!! This test unfortunately cannot change. I Copied your additional test to this change, and added tests for a few other fields I missed: https://github.com/DataDog/datadog-agent/pull/34645